airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalUtils.java

```diff
@@ -37,59 +37,91 @@
+  // This method only exists to allow the migrator to instantiate a cloud and non-cloud temporal client at the same time.
+  public static WorkflowServiceStubs createTemporalService(final boolean isCloud) {
+    if (isCloud) {
+      log.info("createTemporalService chose Cloud...");
+      log.info("Using Temporal Cloud with:\nhost: {}\nnamespace: {}", configs.getTemporalCloudHost(), configs.getTemporalCloudNamespace());
+      try {
+        final InputStream clientCert = new ByteArrayInputStream(configs.getTemporalCloudClientCert().getBytes(StandardCharsets.UTF_8));
+        final InputStream clientKey = new ByteArrayInputStream(configs.getTemporalCloudClientKey().getBytes(StandardCharsets.UTF_8));
+
+        final WorkflowServiceStubsOptions options = WorkflowServiceStubsOptions.newBuilder()
+            .setSslContext(SimpleSslContextBuilder.forPKCS8(clientCert, clientKey).build())
+            .setTarget(configs.getTemporalCloudHost())
+            .build();
+
+        return getTemporalClientWhenConnected(
+            WAIT_INTERVAL,
+            MAX_TIME_TO_CONNECT,
+            WAIT_TIME_AFTER_CONNECT,
+            () -> WorkflowServiceStubs.newInstance(options),
+            configs.getTemporalCloudNamespace());
+      } catch (final Exception e) {
+        throw new RuntimeException(e);
+      }
+    }
+    log.info("createTemporalService chose Airbyte...");
+    log.info("Using Temporal Airbyte with:\nhost: {}\nnamespace: {}", configs.getTemporalHost(), DEFAULT_NAMESPACE);
+    final WorkflowServiceStubsOptions options = WorkflowServiceStubsOptions.newBuilder()
+        .setTarget(configs.getTemporalHost())
+        .build();
 
-  public static void configureTemporalNamespace(final WorkflowServiceStubs temporalService) {
+    final WorkflowServiceStubs temporalService = getTemporalClientWhenConnected(
+        WAIT_INTERVAL,
+        MAX_TIME_TO_CONNECT,
+        WAIT_TIME_AFTER_CONNECT,
+        () -> WorkflowServiceStubs.newInstance(options),
+        DEFAULT_NAMESPACE);
+
+    configureTemporalAirbyteNamespace(temporalService);
+    return temporalService;
+  }
+
+  private static void configureTemporalAirbyteNamespace(final WorkflowServiceStubs temporalService) {
     final var client = temporalService.blockingStub();
     final var describeNamespaceRequest = DescribeNamespaceRequest.newBuilder().setNamespace(DEFAULT_NAMESPACE).build();
     final var currentRetentionGrpcDuration = client.describeNamespace(describeNamespaceRequest).getConfig().getWorkflowExecutionRetentionTtl();
     final var currentRetention = Duration.ofSeconds(currentRetentionGrpcDuration.getSeconds());
 
     if (currentRetention.equals(WORKFLOW_EXECUTION_TTL)) {
-      LOGGER.info("Workflow execution TTL already set for namespace " + DEFAULT_NAMESPACE + ". Remains unchanged as: "
+      log.info("Workflow execution TTL already set for namespace " + DEFAULT_NAMESPACE + ". Remains unchanged as: "
```

Comment:
curious, what is the workflow execution TTL?

Reply:
I believe it's how long we retain workflow history. This is something we're able to configure in our on-prem temporal instance, but not something we're allowed to configure in our Temporal Cloud namespace.